File: connectx-backend/src/main/java/com/connectx/group/dto/UpdateGroupSettingsRequestDto.java
package com.connectx.group.dto;

/**
 * Deliberately carries only the three policy settings themselves -- no groupId, ownerId, actorId,
 * or role. The group id comes from the path, the actor from the authenticated principal, and the
 * actor's authority (docs/CONNECTX_GROUP_ARCHITECTURE.md §6: owner only) is re-derived from the DB
 * by GroupAuthorizationService, never trusted from this request.
 * <p>
 * Every field is optional -- "Request should contain ONLY the settings being changed." A null
 * field is left untouched; only non-null fields are validated and applied, all within one
 * transaction (GroupService#updateSettings), so a request with one invalid field among several
 * valid ones commits none of them.
 */
public class UpdateGroupSettingsRequestDto {

    private String whoCanInvite;
    private String whoCanSendMessages;
    private String whoCanEditGroupInfo;

    public UpdateGroupSettingsRequestDto() {}

    public String getWhoCanInvite() {
        return whoCanInvite;
    }

    public void setWhoCanInvite(String whoCanInvite) {
        this.whoCanInvite = whoCanInvite;
    }

    public String getWhoCanSendMessages() {
        return whoCanSendMessages;
    }

    public void setWhoCanSendMessages(String whoCanSendMessages) {
        this.whoCanSendMessages = whoCanSendMessages;
    }

    public String getWhoCanEditGroupInfo() {
        return whoCanEditGroupInfo;
    }

    public void setWhoCanEditGroupInfo(String whoCanEditGroupInfo) {
        this.whoCanEditGroupInfo = whoCanEditGroupInfo;
    }
}

File: connectx-android/app/src/main/java/com/connectx/app/data/remote/group/model/UpdateGroupSettingsRequestDto.kt
package com.connectx.app.data.remote.group.model

import kotlinx.serialization.Serializable

/**
 * Mirrors com.connectx.group.dto.UpdateGroupSettingsRequestDto exactly. All three
 * policy fields are optional -- request should contain ONLY the settings being
 * changed; a null/omitted field is left untouched server-side. Kept as plain
 * Strings, matching GroupDto's convention for the same three enum-backed fields.
 */
@Serializable
data class UpdateGroupSettingsRequestDto(
    val whoCanInvite: String? = null,
    val whoCanSendMessages: String? = null,
    val whoCanEditGroupInfo: String? = null
)
